### PR TITLE
add mongodb + loan CRUD

### PR DESCRIPTION
**Pull request by shyakadavis**
To test: 
- Make sure you have [MongoDB Compass](https://www.mongodb.com/products/tools/compass) up and running.
- Copy the content of`.env.example` and save it in `.env.local`
- Spin up the dev server,

	```
	pnpm i && pnpm dev
	```
- visit: http://localhost:3000/loans

Note:
- The `loans` route is only for testing purposes. Might/will be deleted later.

### CODE REVIEW

File: src/lib/loan-db.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+			return { error: 'Loan not found' };
+		}
+
+		const loan = await Loan.findById(parsedId).lean().exec();
+		if (loan) {
+			return {
+				loan,
+			};
+		} else {
+			return { error: 'Loan not found' };
```
Comment (abelbordonado) [inline src/lib/loan-db.ts:71]: Return 404

The `else` is not needed because we use return

File: src/app/[locale]/(routes)/loans/loan-form.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	initialThreshold: '',
+	interestRate: '',
+	liquidationThreshold: '',
+	lenderAddress: '',
+	loanRequestDeadline: new Date(),
+	managerNFT: '',
+	managerNFTVersion: '',
+};
+
+export default function LoanForm() {
```
Comment (abelbordonado) [inline src/app/[locale]/(routes)/loans/loan-form.tsx:36]: Put in the readme about this places we are here to test

Why would the metamask not allow you to enter this page?

File: src/lib/loan-db.ts (hunk 1 above)
Comment (shyakadavis) [reply src/lib/loan-db.ts:71]: What do you mean here?

You mean something like:

```ts
const loan = await Loan.findByIdAndUpdate(parsedId, params, { new: true })
			.lean()
			.exec();

		if (!loan) {
			return { error: 'Loan not found' };
		}
		
		return { loan };
```

File: src/app/[locale]/(routes)/loans/loan-form.tsx (hunk 2 above)
Comment (shyakadavis) [reply src/app/[locale]/(routes)/loans/loan-form.tsx:36]: So, a while back in Slack I asked about how I can log in, and what services you are using, but I couldn't find any resources.

Basically, I wanted to know what to do in order to pass the login check, and what services you guys are using/recommend?

**Comment by shyakadavis** [comment]
P.S. The build is failing on Netlify because of this:

```zsh
11:38:52 AM: Failed during stage "building site": Build script returned non-zero exit code: 2
11:38:52 AM: Error: ❌ Please add your MongoDB URI to .env.local
```

Just adding an environment variable of `MONGODB_URI` should fix.

**Pull request merged**